datafusion/physical-expr/src/expressions/not.rs

```diff
@@ -86,10 +86,29 @@ impl PhysicalExpr for NotExpr {
                 )))
             }
             ColumnarValue::Scalar(scalar) => {
-                let bool_value: bool = scalar.try_into()?;
-                Ok(ColumnarValue::Scalar(ScalarValue::Boolean(Some(
-                    !bool_value,
-                ))))
+                match scalar {
+                    ScalarValue::Utf8(utf8_val) => {
+                        // NOT op supports `NOT(NULL)` usage that returns `NULL` directly
+                        if utf8_val.is_none() {
+                            Ok(ColumnarValue::Scalar(ScalarValue::Utf8(None)))
```

Comment:
I think this may be problematic as the output type of `not()` is supposed to be `Boolean`. Does returning

```suggestion
                            Ok(ColumnarValue::Scalar(ScalarValue::Boolean(None)))
```

work?

Reply:
@alamb OK, will fix it in follow on commit